File: Uspokajam-lek-backend/src/main/java/com/example/uspokajamlekbackend/exercise/Exercise.java
package com.example.uspokajamlekbackend.exercise;

import com.example.uspokajamlekbackend.user.doctor.Doctor;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Type;

@Entity //encja, czyli bedzie zapis do bazy danych
@AllArgsConstructor
@RequiredArgsConstructor
@Data //tworzy nam gettery i settery
public class Exercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    @Column(columnDefinition = "VARCHAR(1000)")
    private String description;

    private String duration;
    private int time;
    private String category;

    @ManyToOne
    private Doctor createdBy;

    @Override
    public String toString() {
        return "Exercise [id=" + id + ", name=" + name + ", description=" + description + "]";
    }
}

File: Uspokajam-lek-frontend/src/app/models/exercise.ts
import {Doctor} from "./doctor";

export class Exercise {
  id: number;
  name: string;
  description: string;
  duration: string;
  time: number;
  category: string;
  createdBy: Doctor;


  constructor(id: number, name: string, description: string, duration: string, time: number, category: string, createdBy: Doctor) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.duration = duration;
    this.time = time
    this.category = category;
    this.createdBy = createdBy;
  }
}
